File: IsPalindrome/IsPalindrome.java
package IsPalindrome;

public class IsPalindrome {
    public static void main(String[] args) {
        String s = "abb";
        System.out.println(solution(s));
    }

    public static boolean solution(String s) {
        String sNoSpaces = s.replaceAll("[^A-Za-z0-9]", "").toLowerCase();
        boolean result = false;
        if (sNoSpaces.length() <= 1) {
            result = true;
        } else {
            int start = 0;
            int end = sNoSpaces.length() - 1;
            while (start <= end) {
                if (sNoSpaces.charAt(start) == sNoSpaces.charAt(end)) {
                    result = true;

                } else {
                    result = false;
                    break;
                }

                start++;
                end--;
            }
        }

        return result;
    }
}

File: IsPalindrome/IsPalindrome.py
import re


def isPalindrome(s: str) -> bool:
        sCleaned = re.sub(r'[^A-Za-z0-9]', '', s)
        if(len(sCleaned) <= 1):
            return True
        else:
            
            start = 0
            end = len(sCleaned) - 1
            result = False
            while end >= start:
                if(sCleaned[start].lower() == sCleaned[end].lower()):
                    result = True
                else:
                    result = False
                    break  
                start += 1
                end -= 1
            return result
        
s = "Was it a car or a cat I saw?"
print(isPalindrome(s))
